File: shared/src/main/java/ru/privatenull/pnauth/platform/PlayerResourceKey.java
package ru.privatenull.pnauth.platform;

import java.util.UUID;

/** Shared identity for every named resource owned by a player. */
public final class PlayerResourceKey {
    public final UUID playerId;
    public final String name;

    public PlayerResourceKey(UUID playerId, String resourceId) {
        if (playerId == null) throw new IllegalArgumentException("playerId");
        if (resourceId == null || resourceId.isBlank()) throw new IllegalArgumentException("resourceId");
        this.playerId = playerId;
        this.name = resourceId;
    }

    public UUID playerId() { return playerId; }
    public String resourceId() { return name; }

    @Override public boolean equals(Object other) {
        return this == other || other instanceof PlayerResourceKey key
                && playerId.equals(key.playerId) && name.equals(key.name);
    }

    @Override public int hashCode() { return 31 * playerId.hashCode() + name.hashCode(); }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/platform/PlayerResourceKey.kt
package ru.privatenull.pnauth.platform

import java.util.UUID

/** Shared identity for every named resource owned by a player. */
class PlayerResourceKey(
    @JvmField val playerId: UUID,
    resourceId: String
) {
    @JvmField val name: String = resourceId

    init {
        require(resourceId.isNotBlank()) { "resourceId" }
    }

    fun playerId(): UUID = playerId
    fun resourceId(): String = name

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is PlayerResourceKey) return false
        return playerId == other.playerId && name == other.name
    }

    override fun hashCode(): Int {
        return 31 * playerId.hashCode() + name.hashCode()
    }

    override fun toString(): String {
        return "PlayerResourceKey(playerId=$playerId, name=$name)"
    }
}
